modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectView/ModalAddDefaultSortColumn/ModalAddDefaultSortColumn.tsx

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm from '@clayui/form';
+import ClayModal from '@clayui/modal';
+import React, {
+	FormEvent,
+	useContext,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+
+import AutoComplete from '../../Form/AutoComplete';
+import CustomSelect from '../../Form/CustomSelect/CustomSelect';
+import ViewContext, {TYPES} from '../context';
+import {TObjectViewColumn, TObjectViewSortColumn} from '../types';
+
+interface IProps extends React.HTMLAttributes<HTMLElement> {
+	editingObjectFieldName: string;
+	header: string;
+	isEditingSort: boolean;
+	observer: any;
+	onClose: () => void;
+}
+
+export function ModalAddDefaultSortColumn({
+	editingObjectFieldName,
+	header,
+	isEditingSort,
+	observer,
+	onClose,
+}: IProps) {
+	const [{objectFields, objectView}, dispatch] = useContext(ViewContext);
+
+	const {objectViewColumns, objectViewSortColumns} = objectView;
+	const [availableViewColumns, setAvailableViewColumns] = useState<
+		TObjectViewColumn[]
+	>(objectViewColumns);
+
+	useEffect(() => {
+		const newAvailableViewColumns = objectViewColumns.filter(
+			(viewColumn) => viewColumn.isDefaultSort === false
+		);
+
+		setAvailableViewColumns(newAvailableViewColumns);
+	}, [objectViewColumns]);
+
+	const sortOptions = [
+		{
+			label: Liferay.Language.get('ascending'),
+			value: 'asc',
+		},
+		{
+			label: Liferay.Language.get('descending'),
+			value: 'desc',
+		},
+	];
```

Comment:
Could you put this variable out of this component? Let me know if you don't understand what I mean